client/controllers/graph.controller.js

```diff
@@ -9,111 +9,139 @@
     
     function GraphController(canvasService) {
         var vm = this; // View Model
+        vm.isDrawing = false;
+        vm.isSelecting = false;
+        
+        // Set of vertices in graph - V(G)
+        vm.vertices = [];
+        
+        // Set of edges in graph - E(G)
+        vm.edges = [];
         
         var nextVertexId = 0;
         var nextEdgeId = 0;
         var vertexColor = "#CC8193";
-        vm.canDraw = false;
-        
-        // Set of vertices in graph - V(G)
-        vm.vertices = [
-            {
-                //id: 0, // id of vertex
-                //xPos: 0, // x-position
-                //yPos: 0, // y-position
-                //weight: 0, // vertex weight (if applicable)
-                //size: 0, // size of vertex (radius)
-                //text: '' // vertex name/weight
-            },
-        ];
+        var selectedColor = "#00FFFF";
+        var selectedVertices = [];
         
-        // Set of edges in graph - E(G)
-        vm.edges = [
-            {
-                //v1: 0,
-                //v2: 0
+        // Function: click(event)
+        // handles click events inside the canvas
+        vm.click = function(event) {
+            if(vm.isDrawing) {
+                addVertex(event);
+            } else if(vm.isSelecting) {
+                selectVertex(event);
             }
-        ];
+        };
         
-        // Function: nextChar(currentVertex)
-        // Increments through alphabet to add corresponding
-        // letter on newly inserted verticies
-        vm.nextChar = function(currentVertex){
-            if(currentVertex < 26) {
-                return String.fromCharCode('A'.charCodeAt(0) + currentVertex);    
-            }
-            else {
-                return "[ ]";
-            } 
-        }
+        // Function: toggleDrawing()
+        // toggles the isDrawing switch
+        vm.toggleDrawing = function() {
+            vm.isDrawing = !vm.isDrawing;
+        };
+        
+        // Function: toggleSelecting()
+        // toggles the isSelecting switch
+        vm.toggleSelecting = function() {
+            selectedVertices = [];
+            update();
+            vm.isSelecting = !vm.isSelecting;
+        };
         
         // Function: addVertex(event)
         // adds vertex to V(G)
-        vm.addVertex = function(event) {
-            vm.offsetX = event.offsetX;
-            vm.offsetY = event.offsetY;
-            vm.id = nextVertexId;
+        function addVertex(event) {
+            var offsetX = event.offsetX;
```

Comment:
Changed these expressions to be local variables rather than properties of the view-model since the view-model doesn't really have responsibility of temp data like that.